content/doc/book/pipeline/getting-started.adoc

```diff
@@ -0,0 +1,233 @@
+with the editor that you use in your IDE (integrated development environment) or
+SCM system, and then loading those scripts into Jenkins using the *Pipeline
+Script from SCM* option enabled by the workflow-scm-step plugin, which is one of
+the plugins that the Pipeline plugin depends on and automatically installs.
+
+Loading pipeline scripts using the `checkout scm` step leverages the
+idea of "Pipeline as code" and allows easy maintenance of Pipelines with source
+control and text-editors.
+
+To do this, select *Pipeline script from SCM* when defining the pipeline.
+
+With the *Pipeline script from SCM* option selected, you do not enter any Groovy
+code in the Jenkins UI; you just indicate by specifying a path where in source
+code you want to retrieve the pipeline from. When you update the designated
+repository, a new build is triggered, as long as your job is configured with an
+SCM polling trigger.
+
+[TIP]
+====
+The first line of a `Jenkinsfile` should be `#!groovy`
+footnoteref:[shebang, https://en.wikipedia.org/wiki/Shebang_(Unix)]
+which text editors, IDEs, GitHub, etc will use to syntax highlight the
+`Jenkinsfile` properly as Groovy code.
+====
+
+
+== Built-in Documentation
+
+Pipeline ships with built-in documentation features to make it
+easier to create Pipelines of varying complexities. This built-in documentation
+is automatically generated and updated based on the plugins installed in the
+Jenkins instance.
+
+The built-in documentation can be found globally at:
+link:http://localhost:8080[localhost:8080/pipeline-syntax/],
+assuming you have a Jenkins instance running on localhost port 8080. The same
+documentation is also linked as *Pipeline Syntax* in the side-bar for any
+configured Pipeline project.
+
+image::pipeline-syntax-sidebar.png[Pipeline Syntax in the side-bar, role=center]
+
+[[snippet-generator]]
+=== Snippet Generator
+
+The built-in "Snippet Generator" utility is helpful for creating bits of
+code for individual steps, discovering new steps provided by plugins, or
+experimenting with different parameters for a particular step.
+
+The Snippet Generator is dynamically populated with a list of the steps
+available to the Jenkins instance. The number of steps available is dependent
+on the plugins installed which explicitly expose steps for use in Pipeline.
+
+To generate a step snippet with the Snippet Generator:
+
+. Navigate to the Snippet Generator
+. Select the desired step in the *Sample Step* dropdown menu
+. Use the dynamically populated area below the *Sample Step* dropdown to configure the selected step.
+. Click *Generate Pipeline Script* to create a snippet of Pipeline which can be
+copied and pasted into a Pipeline.
+
+image::snippet-generator.png[Snippet Generator, role=center]
+
+To access additional information and/or documentation about the step selected,
+click on the help icon (blue question mark).
+
+=== Global Variable Reference
```

Comment:
Could we put a TODO here to move the details about specific global variables below to a different section?  This feels a bit odd.